### PR TITLE
Document more scrolling classes

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scroll_activity.dart [hunk 1]
```diff
@@ -106,16 +149,31 @@ class IdleScrollActivity extends ScrollActivity {
...
     ScrollActivityDelegate delegate,
     DragStartDetails details,
     this.onDragCanceled,
-  ) : _delegate = delegate, _lastDetails = details;
+  }) : _delegate = delegate, _lastDetails = details {
+    assert(delegate != null);
+    assert(details != null);
+  }
 
+  /// The object that will acuate the scroll view as the user drags.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: Did you really mean acuate here?

File: packages/flutter/lib/src/widgets/scroll_activity.dart [hunk 2]
```diff
@@ -271,8 +357,8 @@ class BallisticScrollActivity extends ScrollActivity {
 
   /// Move the position to the given location.
   ///
-  /// If the new position was fully applied, return true.
-  /// If there was any overflow, return false.
+  /// If the new position was fully applied, return true. If there was any
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: s/return/returns/?

File: packages/flutter/lib/src/widgets/scroll_activity.dart [hunk 3]
```diff
@@ -336,8 +435,11 @@ class DrivenScrollActivity extends ScrollActivity {
   Completer<Null> _completer;
   AnimationController _controller;
 
+  /// A [Future] that completes when the activity stops.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: Add a discussion that this does not guarantee that the animation will have reached `to` or that the given `duration` will have elapsed.

File: packages/flutter/lib/src/widgets/scroll_context.dart [hunk 4]
```diff
@@ -8,11 +8,34 @@ import 'package:flutter/rendering.dart';
...
+  /// [ScrollBehavior.buildViewportChrome].
   BuildContext get notificationContext;
+
+  /// A [TickerProvider] to use when animating the scroll position.
   TickerProvider get vsync;
+
+  /// The direction in which the widget scrolls.
   AxisDirection get axisDirection;
 
+  /// Whether the widget should ignore [PointerEvent] inputs.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_context.dart]: Specifically this is about whether the entire widget subtree should ignore pointers. It makes the contents of the widget non-interactive.

File: packages/flutter/lib/src/widgets/scroll_activity.dart [hunk 5]
```diff
@@ -106,16 +149,31 @@ class IdleScrollActivity extends ScrollActivity {
...
+///    while a drag is underway.
 class ScrollDragController implements Drag {
-  ScrollDragController(
+  /// Creates an object that scrolls a scroll view as the user drags their
+  /// finger across the screen.
+  ///
+  /// The [delegate] and `details` arguments must not be null.
+  ScrollDragController({
     ScrollActivityDelegate delegate,
     DragStartDetails details,
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: mark them @required?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/widgets/scroll_activity.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: I meant actuate

**Pull request merged**